src/plugins/discord/fetch.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import fetch from "isomorphic-fetch";
+
+import {DepaginatedFetcher, getPages} from "./mirror";
+import {DiscordFetcher, type FetchOptions} from "./fetcher";
+import * as Model from "./models";
+import {type Snowflake} from "./models";
+import Database from "better-sqlite3";
+import {fetchDiscord} from "./mirror";
+import {SqliteMirror} from "./sqliteMirror";
+
+export const DISCORD_SERVER = "https://discordapp.com/api";
+const DISCORD_TOKEN_ENV_NAME = "SOURCECRED_DISCORD_BOT_TOKEN";
+
+function buildDepaginatedFetcher(
+  guildId: Snowflake,
+  discordFetcher: DiscordFetcher
+): DepaginatedFetcher {
+  return {
+    channels(): Promise<$ReadOnlyArray<Model.Channel>> {
+      return discordFetcher.channels(guildId);
+    },
+    members(): Promise<$ReadOnlyArray<Model.GuildMember>> {
+      const getMembers = (after: Snowflake) => {
+        return discordFetcher.members(guildId, after);
+      };
+      return getPages(getMembers, "0");
+    },
+    messages(channel: Snowflake): Promise<$ReadOnlyArray<Model.Message>> {
+      const getMessages = (after: Snowflake) => {
+        return discordFetcher.messages(channel, after);
+      };
+      return getPages(getMessages, "0");
+    },
+    reactions(
+      channel: Snowflake,
+      message: Snowflake,
+      emoji: Model.Emoji
+    ): Promise<$ReadOnlyArray<Model.Reaction>> {
+      const getReactions = (after: Snowflake) => {
+        return discordFetcher.reactions(channel, message, emoji, after);
+      };
+      return getPages(getReactions, "0");
+    },
+  };
+}
+
+function getTokenFromEnv(): string {
+  const token = process.env[DISCORD_TOKEN_ENV_NAME];
+  if (token == null) {
+    throw new Error(
+      `No Discord token provided: please set ${DISCORD_TOKEN_ENV_NAME}`
+    );
+  }
+  return token;
+}
+
+export function buildDiscordFetch(fetch: typeof fetch) {
+  return async (endpoint: string) => {
+    const fetchOptions = {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        Authorization: `Bot ${getTokenFromEnv()}`,
```

Comment:
Let's not bind in the token here as an impl detail--instead, let's have the caller pass the token in. In the future we may get tokens from different places than the environment (e.g. a config store) so we don't want to hard code this assumption.